src/librustc_driver/driver.rs

```diff
@@ -73,6 +74,20 @@ pub fn compile_input(trans: Box<TransCrate>,
                      output: &Option<PathBuf>,
                      addl_plugins: Option<Vec<String>>,
                      control: &CompileController) -> CompileResult {
+    PROFQ_CHAN.set(&sess.profile_channel, || {
+        compile_input_impl(trans, sess, cstore, input_path, input, outdir, output, addl_plugins, control)
+    })
+}
```

Comment:
Note that this is the point were the Rayon thread pool is created in my branch. This is split out to its own function to avoid rightward drift (and merge conflicts).